### ISSUE TITLE
AWS CLI wellarchitected list-lens-review-improvements miss ImprovementPlans in response 

### ISSUE THREAD

**Issue by MaximYesman** [aws/aws-cli#7347]
### Describe the bug

When running `aws wellarchitected list-lens-review-improvements...` - `ImprovementPlans` field under the `ImprovementSummaries` is missed 

### Expected Behavior

When running `aws wellarchitected list-lens-review-improvements...` - `ImprovementPlans` field under the `ImprovementSummaries` should present and contains the same data, as seen in the console > workload > finished lens > "Improvement plan" tab > "Improvement items" list > "Recommended improvement items" section 

### Current Behavior

When running `aws wellarchitected list-lens-review-improvements...` - `ImprovementPlans` field under the `ImprovementSummaries` is missed 

### Reproduction Steps

Run `aws wellarchitected list-lens-review-improvements...` 
The result doesn't have`ImprovementPlans` field under the `ImprovementSummaries`

### CLI version used

aws-cli/2.2.31 Python/3.8.8 Darwin/21.6.0 exe/x86_64 prompt/off

### Environment details (OS name and version, etc.)

macOS Monterey, Version 12.6

**Comment by MaximYesman** [comment]
`wellarchitected list-lens-review-improvements` CLI command documentation https://awscli.amazonaws.com/v2/documentation/api/latest/reference/wellarchitected/list-lens-review-improvements.html

**Comment by aBurmeseDev** [comment]
Hi @MaximYesman, thanks for reaching out.

It looks like you're using older version of CLI, could you update to the latest version ([2.8.2](https://github.com/aws/aws-cli/blob/v2/CHANGELOG.rst)) and try running the command?

Here's [docs guide](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html) on how to update CLI for reference.

Best,
John

**Comment by MaximYesman** [comment]
Hello!
Thank you for fast response!

CLI was updated to version: aws-cli/2.8.3 Python/3.9.11 Darwin/21.6.0 exe/x86_64 prompt/off
Now I can see `ImprovementPlans` field, but it empty for each record, but in workload I have improvement plans.
<img width="552" alt="Screenshot 2022-10-19 at 09 37 20" src="https://user-images.githubusercontent.com/15417876/196626762-81c73aaf-6a03-40bd-9244-a08be3532837.png">
<img width="1098" alt="Screenshot 2022-10-19 at 09 38 55" src="https://user-images.githubusercontent.com/15417876/196627127-74a54f13-36b6-4e32-9485-820cf1ecbe6d.png">

Note: same behaviour when using AWS Java SDK

**Comment by MaximYesman** [comment]
Hello, @aBurmeseDev!
Any updates?

**Comment by aBurmeseDev** [comment]
Hi @MaximYesman, sorry for the delay. I reached out to service API team and will be following up for updates. Could you please share debug logs (with any sensitive info redacted) by adding `--debug` to the command while we wait for an update? It will give us more insight on what the issue is. Thank you for your patience!

**Comment by MaximYesman** [comment]
Thank you for the update!
Here is a debug output:

```2022-10-26 09:42:15,968 - MainThread - awscli.clidriver - DEBUG - CLI version: aws-cli/2.8.3 Python/3.9.11 Darwin/21.6.0 exe/x86_64
2022-10-26 09:42:15,968 - MainThread - awscli.clidriver - DEBUG - Arguments entered to CLI: ['wellarchitected', 'list-lens-review-improvements', '--workload-id', 'XXXXX', '--lens-alias', 'wellarchitected', '--profile', 'waf', '--region', 'us-east-1', '--debug']
2022-10-26 09:42:15,989 - MainThread - botocore.hooks - DEBUG - Event building-command-table.main: calling handler <function add_s3 at 0x7ff6b84f34c0>
2022-10-26 09:42:15,989 - MainThread - botocore.hooks - DEBUG - Event building-command-table.main: calling handler <function add_ddb at 0x7ff6b844bc10>
2022-10-26 09:42:15,989 - MainThread - botocore.hooks - DEBUG - Event building-command-table.main: calling handler <bound method BasicCommand.add_command of <class 'awscli.customizations.configure.configure.ConfigureCommand'>>
2022-10-26 09:42:15,989 - MainThread - botocore.hooks - DEBUG - Event building-command-table.main: calling handler <function change_name at 0x7ff6a8352ca0>
2022-10-26 09:42:15,990 - MainThread - botocore.hooks - DEBUG - Event building-command-table.main: calling handler <function change_name at 0x7ff6a835aca0>
2022-10-26 09:42:15,990 - MainThread - botocore.hooks - DEBUG - Event building-command-table.main: calling handler <function alias_opsworks_cm at 0x7ff6b8504e50>
2022-10-26 09:42:15,990 - MainThread - botocore.hooks - DEBUG - Event building-command-table.main: calling handler <function add_history_commands at 0x7ff6782e0790>
2022-10-26 09:42:15,990 - MainThread - botocore.hooks - DEBUG - Event building-command-table.main: calling handler <bound method BasicCommand.add_command of <class 'awscli.customizations.devcommands.CLIDevCommand'>>
2022-10-26 09:42:15,990 - MainThread - botocore.hooks - DEBUG - Event building-command-table.main: calling handler <function add_waiters at 0x7ff6b85040d0>
2022-10-26 09:42:15,990 - MainThread - botocore.loaders - DEBUG - Loading JSON file: /usr/local/aws-cli/awscli/data/cli.json
2022-10-26 09:42:15,992 - MainThread - botocore.hooks - DEBUG - Event top-level-args-parsed: calling handler <function resolve_types at 0x7ff6783941f0>
2022-10-26 09:42:15,992 - MainThread - botocore.hooks - DEBUG - Event top-level-args-parsed: calling handler <function no_sign_request at 0x7ff678394d30>
2022-10-26 09:42:15,992 - MainThread - botocore.hooks - DEBUG - Event top-level-args-parsed: calling handler <function resolve_verify_ssl at 0x7ff678394ca0>
2022-10-26 09:42:15,992 - MainThread - botocore.hooks - DEBUG - Event top-level-args-parsed: calling handler <function resolve_cli_read_timeout at 0x7ff678394e50>
2022-10-26 09:42:15,992 - MainThread - botocore.hooks - DEBUG - Event top-level-args-parsed: calling handler <function resolve_cli_connect_timeout at 0x7ff678394dc0>
2022-10-26 09:42:15,992 - MainThread - botocore.hooks - DEBUG - Event top-level-args-parsed: calling handler <built-in method update of dict object at 0x7ff6d82cd200>
2022-10-26 09:42:15,992 - MainThread - botocore.session - DEBUG - Setting config variable for profile to 'waf'
2022-10-26 09:42:15,992 - MainThread - botocore.session - DEBUG - Setting config variable for region to 'us-east-1'
2022-10-26 09:42:15,993 - MainThread - awscli.clidriver - DEBUG - CLI version: aws-cli/2.8.3 Python/3.9.11 Darwin/21.6.0 exe/x86_64 prompt/off
2022-10-26 09:42:15,993 - MainThread - awscli.clidriver - DEBUG - Arguments entered to CLI: ['wellarchitected', 'list-lens-review-improvements', '--workload-id', 'XXXXX', '--lens-alias', 'wellarchitected', '--profile', 'waf', '--region', 'us-east-1', '--debug']
2022-10-26 09:42:15,993 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function add_timestamp_parser at 0x7ff6b84f3af0>
2022-10-26 09:42:15,993 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function register_uri_param_handler at 0x7ff6d822a0d0>
2022-10-26 09:42:15,993 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function add_binary_formatter at 0x7ff6d828ff70>
2022-10-26 09:42:15,993 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function no_pager_handler at 0x7ff678274550>
2022-10-26 09:42:15,993 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function inject_assume_role_provider_cache at 0x7ff6d824dc10>
2022-10-26 09:42:15,995 - MainThread - botocore.utils - DEBUG - IMDS ENDPOINT: http://169.254.169.254/
2022-10-26 09:42:16,000 - MainThread - botocore.credentials - DEBUG - Skipping environment variable credential check because profile name was explicitly set.
2022-10-26 09:42:16,000 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function attach_history_handler at 0x7ff6782e0670>
2022-10-26 09:42:16,000 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function inject_json_file_cache at 0x7ff6b8448af0>
2022-10-26 09:42:16,021 - MainThread - botocore.loaders - DEBUG - Loading JSON file: /usr/local/aws-cli/awscli/botocore/data/wellarchitected/2020-03-31/service-2.json
2022-10-26 09:42:16,027 - MainThread - botocore.hooks - DEBUG - Event building-command-table.wellarchitected: calling handler <function add_waiters at 0x7ff6b85040d0>
2022-10-26 09:42:16,047 - MainThread - awscli.clidriver - DEBUG - OrderedDict([('workload-id', <awscli.arguments.CLIArgument object at 0x7ff688729880>), ('lens-alias', <awscli.arguments.CLIArgument object at 0x7ff6887297c0>), ('pillar-id', <awscli.arguments.CLIArgument object at 0x7ff688729790>), ('milestone-number', <awscli.arguments.CLIArgument object at 0x7ff688729730>), ('next-token', <awscli.arguments.CLIArgument object at 0x7ff688729700>), ('max-results', <awscli.arguments.CLIArgument object at 0x7ff6887296d0>)])
2022-10-26 09:42:16,048 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.wellarchitected.list-lens-review-improvements: calling handler <function add_streaming_output_arg at 0x7ff6b84f3dc0>
2022-10-26 09:42:16,048 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.wellarchitected.list-lens-review-improvements: calling handler <function add_cli_input_json at 0x7ff6d8253430>
2022-10-26 09:42:16,048 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.wellarchitected.list-lens-review-improvements: calling handler <function add_cli_input_yaml at 0x7ff6d8253670>
2022-10-26 09:42:16,048 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.wellarchitected.list-lens-review-improvements: calling handler <function unify_paging_params at 0x7ff6b84581f0>
2022-10-26 09:42:16,067 - MainThread - botocore.loaders - DEBUG - Loading JSON file: /usr/local/aws-cli/awscli/botocore/data/wellarchitected/2020-03-31/paginators-1.json
2022-10-26 09:42:16,067 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.wellarchitected.list-lens-review-improvements: calling handler <function add_generate_skeleton at 0x7ff678389790>
2022-10-26 09:42:16,067 - MainThread - botocore.hooks - DEBUG - Event before-building-argument-table-parser.wellarchitected.list-lens-review-improvements: calling handler <bound method OverrideRequiredArgsArgument.override_required_args of <awscli.customizations.cliinput.CliInputJSONArgument object at 0x7ff688729940>>
2022-10-26 09:42:16,067 - MainThread - botocore.hooks - DEBUG - Event before-building-argument-table-parser.wellarchitected.list-lens-review-improvements: calling handler <bound method OverrideRequiredArgsArgument.override_required_args of <awscli.customizations.cliinput.CliInputYAMLArgument object at 0x7ff6887299d0>>
2022-10-26 09:42:16,067 - MainThread - botocore.hooks - DEBUG - Event before-building-argument-table-parser.wellarchitected.list-lens-review-improvements: calling handler <bound method GenerateCliSkeletonArgument.override_required_args of <awscli.customizations.generatecliskeleton.GenerateCliSkeletonArgument object at 0x7ff688729b50>>
2022-10-26 09:42:16,068 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.wellarchitected.list-lens-review-improvements.workload-id: calling handler <awscli.paramfile.URIArgumentHandler object at 0x7ff6d8311fa0>
2022-10-26 09:42:16,068 - MainThread - botocore.hooks - DEBUG - Event process-cli-arg.wellarchitected.list-lens-review-improvements: calling handler <awscli.argprocess.ParamShorthandParser object at 0x7ff67829c9d0>
2022-10-26 09:42:16,068 - MainThread - awscli.arguments - DEBUG - Unpacked value of 'XXXXX' for parameter "workload_id": 'XXXXX'
2022-10-26 09:42:16,068 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.wellarchitected.list-lens-review-improvements.lens-alias: calling handler <awscli.paramfile.URIArgumentHandler object at 0x7ff6d8311fa0>
2022-10-26 09:42:16,068 - MainThread - botocore.hooks - DEBUG - Event process-cli-arg.wellarchitected.list-lens-review-improvements: calling handler <awscli.argprocess.ParamShorthandParser object at 0x7ff67829c9d0>
2022-10-26 09:42:16,068 - MainThread - awscli.arguments - DEBUG - Unpacked value of 'wellarchitected' for parameter "lens_alias": 'wellarchitected'
2022-10-26 09:42:16,068 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.wellarchitected.list-lens-review-improvements.pillar-id: calling handler <awscli.paramfile.URIArgumentHandler object at 0x7ff6d8311fa0>
2022-10-26 09:42:16,068 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.wellarchitected.list-lens-review-improvements.milestone-number: calling handler <awscli.paramfile.URIArgumentHandler object at 0x7ff6d8311fa0>
2022-10-26 09:42:16,068 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.wellarchitected.list-lens-review-improvements.next-token: calling handler <awscli.paramfile.URIArgumentHandler object at 0x7ff6d8311fa0>
2022-10-26 09:42:16,068 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.wellarchitected.list-lens-review-improvements.max-results: calling handler <awscli.paramfile.URIArgumentHandler object at 0x7ff6d8311fa0>
2022-10-26 09:42:16,068 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.wellarchitected.list-lens-review-improvements.cli-input-json: calling handler <awscli.paramfile.URIArgumentHandler object at 0x7ff6d8311fa0>
2022-10-26 09:42:16,068 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.wellarchitected.list-lens-review-improvements.cli-input-yaml: calling handler <awscli.paramfile.URIArgumentHandler object at 0x7ff6d8311fa0>
2022-10-26 09:42:16,068 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.wellarchitected.list-lens-review-improvements.generate-cli-skeleton: calling handler <awscli.paramfile.URIArgumentHandler object at 0x7ff6d8311fa0>
2022-10-26 09:42:16,068 - MainThread - botocore.hooks - DEBUG - Event calling-command.wellarchitected.list-lens-review-improvements: calling handler <bound method CliInputArgument.add_to_call_parameters of <awscli.customizations.cliinput.CliInputJSONArgument object at 0x7ff688729940>>
2022-10-26 09:42:16,069 - MainThread - botocore.hooks - DEBUG - Event calling-command.wellarchitected.list-lens-review-improvements: calling handler <bound method CliInputArgument.add_to_call_parameters of <awscli.customizations.cliinput.CliInputYAMLArgument object at 0x7ff6887299d0>>
2022-10-26 09:42:16,069 - MainThread - botocore.hooks - DEBUG - Event calling-command.wellarchitected.list-lens-review-improvements: calling handler <bound method GenerateCliSkeletonArgument.generate_skeleton of <awscli.customizations.generatecliskeleton.GenerateCliSkeletonArgument object at 0x7ff688729b50>>
2022-10-26 09:42:16,069 - MainThread - botocore.credentials - DEBUG - Looking for credentials via: assume-role
2022-10-26 09:42:16,069 - MainThread - botocore.credentials - DEBUG - Looking for credentials via: assume-role-with-web-identity
2022-10-26 09:42:16,069 - MainThread - botocore.credentials - DEBUG - Looking for credentials via: sso
2022-10-26 09:42:16,069 - MainThread - botocore.credentials - DEBUG - Looking for credentials via: shared-credentials-file
2022-10-26 09:42:16,069 - MainThread - botocore.credentials - INFO - Found credentials in shared credentials file: ~/.aws/credentials
2022-10-26 09:42:16,070 - MainThread - botocore.loaders - DEBUG - Loading JSON file: /usr/local/aws-cli/awscli/botocore/data/endpoints.json
2022-10-26 09:42:16,083 - MainThread - botocore.hooks - DEBUG - Event choose-service-name: calling handler <function handle_service_name_alias at 0x7ff698186430>
2022-10-26 09:42:16,084 - MainThread - botocore.hooks - DEBUG - Event creating-client-class.wellarchitected: calling handler <function add_generate_presigned_url at 0x7ff6c81df1f0>
2022-10-26 09:42:16,088 - MainThread - botocore.endpoint - DEBUG - Setting wellarchitected timeout as (60, 60)
2022-10-26 09:42:16,088 - MainThread - botocore.hooks - DEBUG - Event provide-client-params.wellarchitected.ListLensReviewImprovements: calling handler <function base64_decode_input_blobs at 0x7ff6d8290700>
2022-10-26 09:42:16,088 - MainThread - botocore.hooks - DEBUG - Event before-parameter-build.wellarchitected.ListLensReviewImprovements: calling handler <function generate_idempotent_uuid at 0x7ff6981a0550>
2022-10-26 09:42:16,089 - MainThread - botocore.hooks - DEBUG - Event before-call.wellarchitected.ListLensReviewImprovements: calling handler <function inject_api_version_header_if_needed at 0x7ff6981a6dc0>
2022-10-26 09:42:16,089 - MainThread - botocore.endpoint - DEBUG - Making request for OperationModel(name=ListLensReviewImprovements) with params: {'url_path': '/workloads/XXXXX/lensReviews/wellarchitected/improvements', 'query_string': {}, 'method': 'GET', 'headers': {'User-Agent': 'aws-cli/2.8.3 Python/3.9.11 Darwin/21.6.0 exe/x86_64 prompt/off command/wellarchitected.list-lens-review-improvements'}, 'body': b'', 'url': 'https://wellarchitected.us-east-1.amazonaws.com/workloads/XXXXX/lensReviews/wellarchitected/improvements', 'context': {'client_region': 'us-east-1', 'client_config': <botocore.config.Config object at 0x7ff6784c2160>, 'has_streaming_input': False, 'auth_type': None}}
2022-10-26 09:42:16,089 - MainThread - botocore.hooks - DEBUG - Event request-created.wellarchitected.ListLensReviewImprovements: calling handler <bound method RequestSigner.handler of <botocore.signers.RequestSigner object at 0x7ff6784c2250>>
2022-10-26 09:42:16,089 - MainThread - botocore.hooks - DEBUG - Event choose-signer.wellarchitected.ListLensReviewImprovements: calling handler <function set_operation_specific_signer at 0x7ff6981a0430>
2022-10-26 09:42:16,089 - MainThread - botocore.auth - DEBUG - Calculating signature using v4 auth.
2022-10-26 09:42:16,089 - MainThread - botocore.auth - DEBUG - CanonicalRequest:
GET
/workloads/XXXXX/lensReviews/wellarchitected/improvements

host:wellarchitected.us-east-1.amazonaws.com
x-amz-date:20221026T074216Z
x-amz-security-token:"XXXXX"

host;x-amz-date;x-amz-security-token
XXXXX
2022-10-26 09:42:16,089 - MainThread - botocore.auth - DEBUG - StringToSign:
AWS4-HMAC-SHA256
20221026T074216Z
20221026/us-east-1/wellarchitected/aws4_request
XXXXX
2022-10-26 09:42:16,090 - MainThread - botocore.auth - DEBUG - Signature:
XXXXX
2022-10-26 09:42:16,090 - MainThread - botocore.endpoint - DEBUG - Sending http request: <AWSPreparedRequest stream_output=False, method=GET, url=https://wellarchitected.us-east-1.amazonaws.com/workloads/XXXXX/lensReviews/wellarchitected/improvements, headers={'User-Agent': b'aws-cli/2.8.3 Python/3.9.11 Darwin/21.6.0 exe/x86_64 prompt/off command/wellarchitected.list-lens-review-improvements', 'X-Amz-Date': b'20221026T074216Z', 'X-Amz-Security-Token': b'"XXXXX"', 'Authorization': b'AWS4-HMAC-SHA256 Credential=AXXXXX, SignedHeaders=host;x-amz-date;x-amz-security-token, Signature=XXXXX'}>
2022-10-26 09:42:16,090 - MainThread - botocore.httpsession - DEBUG - Certificate path: /usr/local/aws-cli/awscli/botocore/cacert.pem
2022-10-26 09:42:16,090 - MainThread - urllib3.connectionpool - DEBUG - Starting new HTTPS connection (1): wellarchitected.us-east-1.amazonaws.com:443
2022-10-26 09:42:16,758 - MainThread - urllib3.connectionpool - DEBUG - https://wellarchitected.us-east-1.amazonaws.com:443 "GET /workloads/XXXXX/lensReviews/wellarchitected/improvements HTTP/1.1" 200 12238
2022-10-26 09:42:16,758 - MainThread - botocore.parsers - DEBUG - Response headers: {'Date': 'Wed, 26 Oct 2022 07:42:16 GMT', 'Content-Type': 'application/json', 'Content-Length': '12238', 'Connection': 'keep-alive', 'x-amzn-RequestId': '183f5ecf-edb0-41b3-aa11-7f67b1d86def', 'Access-Control-Allow-Origin': '*', 'Access-Control-Allow-Headers': 'authorization,content-type,x-amz-content-sha256,x-amz-date,x-amz-security-token,x-amz-user-agent', 'x-amz-apigw-id': 'XXXXX', 'Access-Control-Expose-Headers': 'x-amzn-errortype,x-amzn-requestid,x-amzn-errormessage,x-amzn-trace-id,x-amzn-requestid,x-amz-apigw-id,date', 'X-Amzn-Trace-Id': 'Root=XXXXX'}
2022-10-26 09:42:16,759 - MainThread - botocore.parsers - DEBUG - Response body:
b' {\n    "WorkloadId": "XXXXX",\n    "LensArn": "XXXXX",\n    "LensAlias": "wellarchitected",\n         "NextToken": "XXXXX",     "ImprovementSummaries": [\n                                                        {\n                    "QuestionId": "priorities",\n                    "QuestionTitle":  "How do you determine what your priorities are?",\n                                                                                                                         "Risk": "HIGH",                                                                                     "ImprovementPlanUrl": "https://wa.aws.amazon.com/TypeII/en/wellarchitected/wellarchitected.priorities.improvement-plan.en.html",\n                                        "PillarId": "operationalExcellence",\n                    "ImprovementPlans": [\n                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    ]\n                }\n                ,                                                {\n                    "QuestionId": "ready-to-support",\n                    "QuestionTitle":  "How do you know that you are ready to support a workload?",\n                                                                                                                         "Risk": "HIGH",                                                                                     "ImprovementPlanUrl": "https://wa.aws.amazon.com/TypeII/en/wellarchitected/wellarchitected.ready-to-support.improvement-plan.en.html",\n                                        "PillarId": "operationalExcellence",\n                    "ImprovementPlans": [\n                                                                                                                                                                                                                                                                                                                                                                                            ]\n                }....]\n}'
2022-10-26 09:42:16,760 - MainThread - botocore.hooks - DEBUG - Event needs-retry.wellarchitected.ListLensReviewImprovements: calling handler <bound method RetryHandler.needs_retry of <botocore.retries.standard.RetryHandler object at 0x7ff6784c2ca0>>
2022-10-26 09:42:16,760 - MainThread - botocore.retries.standard - DEBUG - Not retrying request.
2022-10-26 09:42:16,760 - MainThread - botocore.hooks - DEBUG - Event after-call.wellarchitected.ListLensReviewImprovements: calling handler <bound method RetryQuotaChecker.release_retry_quota of <botocore.retries.standard.RetryQuotaChecker object at 0x7ff6784c26d0>>
2022-10-26 09:42:16,760 - MainThread - awscli.formatter - DEBUG - RequestId: 183f5ecf-edb0-41b3-aa11-7f67b1d86def```

**Comment by aBurmeseDev** [comment]
@MaximYesman thanks for providing the logs and being patient with us. I just heard back from AWS WellArchitected Team and here's their input:
>The improvement plans that you see in the console actually come from the value in the field "ImprovementPlanUrl" and not from the field "improvementPlans" .
The field improvementPlans was added to allow customers to add Improvement Plans at the best practice level only for the **Custom Lens** that they create .
That being said, this is not a bug but I do agree that this makes it confusing from a customer standpoint, and we will look into ways to disambiguate it for customers.

To summarize, this was expected behavior unless you're using Custom Lens in your case and I am sorry for the confusion caused. Please let me know if you have any other questions!

**Comment by MaximYesman** [comment]
Thank you for the update and detailed description, @aBurmeseDev 
The issue is quite confusing, but it can be closed, I think

But still, can you advise any API, which will return me the list of all best practices from the improvement plan of the completed workload? As on a screenshot that I sent at the beginning of the conversation.

Or only option is to call [list-answers](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/wellarchitected/list-answers.html) and map answers to [description pages](https://wa.aws.amazon.com/wat.map.en.html) using name conventions?

**Comment by aBurmeseDev** [comment]
@MaximYesman I reached out to the API team regarding your follow up question to return a list of all best practices from the improvement plan of the completed workload. I will post the update as soon as I hear back. Thanks for your patience!

**Comment by aBurmeseDev** [comment]
Hi @MaximYesman, I have heard back from API team and unfortunately, wellarchitected API today does not return a list of improvement plans as It only returns a URL which could contain multiple improvement plans. Sorry for the inconvenience. 

Please let us know if you have any other questions.
